### ISSUE TITLE
Update topology fails for a non-acking topology under back-pressure

### ISSUE THREAD

**Issue by ashvina** [twitter/heron#1690]
I am observing a confusing behavior with update feature. I have a two-stage topology with a slow bolt. The slow bolt starts a back-pressure. I tried the update command to increase the number of bolt instances. The update command failed. It seems the `TMaster` never created a physical plan again. This behavior is not applicable to a acking topology. 

To reproduce add a `200 millisec` delay in the`execute(Tuple tuple)` method  for bolts in `ExclamationTopology` and `AckingTopology`. After sometime the `exclaim1` component starts a backpressure (e.g. metric value for `__time_spent_back_pressure_by_compid/container_1_exclaim1_1` is 60003). Now try updating the `exclaim1` component, `--component-parallelism exclaim1:3`.

I observe that update of the `AckingTopology` is successful. The `ExclamationTopology` update fails after 10 mins with this repeated message: 
> com.twitter.heron.scheduler.UpdateTopologyManager INFO:  Couldn't fetch physical plan for topology ET.   This is probably because stream managers are still registering with TMaster. Will sleep and try again

And there is no way to update a non-acking topology without killing the topology?

**Comment by billonahill** [comment]
@ashvina during an update the physical plan gets removed from the state manager, then the TMaster awaits registration from the existing and new stream managers before updating the physical plan. In this case one or more stream managers never registered. The way to troubleshoot is to identify which one had problems by looking in the tmaster logs, then checking the logs of that stream manager to understand why it failed to register.

**Comment by ashvina** [comment]
@billonahill Thanks for the suggestions. I will take a look at the logs and share what I see.

I am assuming this is a bug? I.e. the stream manager should not wait for the buffers to drain and the tuples in transit will be lost for non-acking topologies.

**Comment by billonahill** [comment]
Correct. All stream managers are restarted actually as part of update.
